subsys/mgmt/mcumgr/smp_pkasm.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <net/buf.h>
+#include <net/net_ip.h>
+#include <mgmt/mcumgr/buf.h>
+#include <mgmt/mcumgr/smp.h>
+#include <mgmt/mgmt.h>
+#include <smp/smp.h>
+
+int zephyr_smp_pkasm_collect(struct zephyr_smp_transport *zst, const void *buf, uint16_t len)
+{
+	/* No allocated buffer indicates that we are expecting first fragment */
+	if (zst->_pkasm.current == NULL) {
```

Comment:
Please unify the style of NULL checks in this PR (either _!zst->_pkasm.current_ or like in the line above).

Reply:
> Please unify the style of NULL checks in this PR (either _!zst->_pkasm.current_ or like in the line above).

I do not think that I have used `!p` where `p` is pointer in any comparison. The NULL, according to C standard is  implementation specific, so it does not have to be 0, which means that neither `if(p)` or `if(!p)` is correct.
Functions returning pointers should either return valid pointer or, which is not recommended by CERT (ERR02-C: Avoid in-band error indicators), NULL, so the comparison should be against NULL.